src/extra/others/controller/lv_obj_controller.c

```diff
@@ -0,0 +1,372 @@
+{
+    size_t size = 0;
+    for(manager_stack_t * cur = manager->top; cur; cur = cur->prev) {
+        size++;
+    }
+    return size;
+}
+
+void lv_controller_manager_pop(lv_controller_manager_t * manager)
+{
+    LV_ASSERT(manager);
+    manager_stack_t * top = manager->top;
+    if(!top) return;
+    manager_stack_t * prev = top->prev;
+    bool dialog = top->dialog;
+    if(!dialog && prev) {
+        item_create_controller(manager, prev, NULL);
+    }
+    item_destroy_obj(manager, top);
+    item_destroy_controller(top);
+    lv_mem_free(top);
+    if(!dialog && prev) {
+        item_create_obj(manager, prev, manager->container, NULL);
+    }
+    manager->top = prev;
+}
+
+bool lv_controller_manager_dispatch_event(lv_controller_manager_t * manager, int which, void * data1, void * data2)
+{
+    LV_ASSERT(manager);
+    manager_stack_t * top = manager->top;
+    if(!top) return false;
+    lv_obj_controller_t * controller = top->controller;
+    if(!controller || !top->cls->event_cb) return false;
+    return top->cls->event_cb(controller, which, data1, data2);
+}
+
+#if LV_USE_MSGBOX
+
+void lv_controller_manager_show(lv_controller_manager_t * manager, const lv_obj_controller_class_t * cls, void * args)
+{
+    LV_ASSERT(manager);
+    LV_ASSERT(cls);
+    manager_stack_t * item = item_new(cls);
+    item_create_controller(manager, item, args);
+    item_create_obj(manager, item, NULL, &lv_msgbox_class);
+    item->dialog = true;
+    manager_stack_t * top = manager->top;
+    item->prev = top;
+    manager->top = item;
+}
+
+#endif
+
+lv_obj_controller_t * lv_controller_manager_top_controller(lv_controller_manager_t * manager)
+{
+    LV_ASSERT(manager);
+    manager_stack_t * top = manager->top;
+    if(!top)return NULL;
+    return top->controller;
+}
+
+lv_obj_controller_t * lv_controller_manager_parent(lv_controller_manager_t * manager)
+{
+    LV_ASSERT(manager);
+    return manager->parent;
+}
+
+lv_obj_controller_t * lv_obj_controller_class_create_unmanaged(const lv_obj_controller_class_t * cls, lv_obj_t * parent,
+                                                               void * args)
+{
+    LV_ASSERT(cls);
+    LV_ASSERT(cls->instance_size);
+    LV_ASSERT(cls->create_obj_cb);
+    lv_obj_controller_t * controller = lv_mem_alloc(cls->instance_size);
+    lv_memset_00(controller, cls->instance_size);
+    controller->cls = cls;
+    if(cls->constructor_cb) {
+        cls->constructor_cb(controller, args);
+    }
+    lv_obj_t * obj = cls->create_obj_cb(controller, parent);
+    controller->obj = obj;
+    if(cls->obj_created_cb) {
+        cls->obj_created_cb(controller, obj);
+    }
+    return controller;
+}
+
+void lv_obj_controller_pop(lv_obj_controller_t * controller)
+{
+    LV_ASSERT(controller);
+    lv_controller_manager_t * manager = controller->manager;
+    LV_ASSERT(manager);
+    LV_ASSERT(manager->top->controller == controller);
+    lv_controller_manager_pop(manager);
```

Comment:
I found a names a little bit incontinent and hard to follow in some places.
- `manager` and `controller` have a similar meaning. It's hard to imagine what is difference
- Some functions start with `lv_controller` and other with `lv_obj_controller`. It should be identical to the name of the file.

Reply:
* What about replacing `manager` with `navigator`?
* `lv_controller_manager_` will be changed to `lv_controller_navigator` then.

Reply:
I'm not sure I fully understand the roles of manager and controller. `manager` can replace, push or pop `controller`s, right?

Reply:
Yes. `manager` is a stack to manage controllers.

Reply:
I see, slowly I'm starting to wrap my head around this concept. :slightly_smiling_face:  It seems really useful to manage life cycles and deal with states. 

I think the word `manager` is fine, but `controller` is strange to me. What does it control? In Android it's "Fragment", isn't it? I think it's describes the purpose better, as it's really a UI fragment. 

I'm not sure we should steal this word (maybe we should), but this naming approach seems better to me.

Reply:
I'm OK with fragment naming thing. So let's rename it to fragment and fragment_manager.

Reply:
Great!